include/sys/stat.h

```diff
@@ -0,0 +1,65 @@
+/*                                                                           */
+/*                   Constants for the mode argument of open                 */
+/*                                                                           */
+/*                                                                           */
+/*                                                                           */
+/* (C) 2003      Ullrich von Bassewitz                                       */
+/*               R�merstrasse 52                                             */
+/*               D-70794 Filderstadt                                         */
+/* EMail:        uz@cc65.org                                                 */
+/*                                                                           */
+/*                                                                           */
+/* This software is provided 'as-is', without any expressed or implied       */
+/* warranty.  In no event will the authors be held liable for any damages    */
+/* arising from the use of this software.                                    */
+/*                                                                           */
+/* Permission is granted to anyone to use this software for any purpose,     */
+/* including commercial applications, and to alter it and redistribute it    */
+/* freely, subject to the following restrictions:                            */
+/*                                                                           */
+/* 1. The origin of this software must not be misrepresented; you must not   */
+/*    claim that you wrote the original software. If you use this software   */
+/*    in a product, an acknowledgment in the product documentation would be  */
+/*    appreciated but is not required.                                       */
+/* 2. Altered source versions must be plainly marked as such, and must not   */
+/*    be misrepresented as being the original software.                      */
+/* 3. This notice may not be removed or altered from any source              */
+/*    distribution.                                                          */
+/*                                                                           */
+/*****************************************************************************/
+
+
+
+#ifndef _STAT_H
+#define _STAT_H
+
+
+
+/*****************************************************************************/
+/*                                   Data                                    */
+/*****************************************************************************/
+
+
+
+/* These are the values for the traditional UNIX mode bits:
+** https://en.wikipedia.org/wiki/File_system_permissions#Numeric_notation
+** (S_IREAD and S_IWRITE are aliases for S_IRUSR and S_IWUSR)
+**
+** Must match the values in asminc/stat.inc and src/sim65/paravirt.c
+*/
+
+#define S_IREAD  0400
```

Comment:
I'm thinking about these values...

Their type: cc65 doesn't define a mode_t. But it says `int __fastcall__ creat (const char* name, unsigned mode);` so `mode` is "known" to be `unsigned`.

Their values: I don't think it's beneficial to use UNIX values here. It sort of implies that things work like on UNIX - which of course is wrong. If you look at
```
#define O_RDONLY        0x01
#define O_WRONLY        0x02
#define O_RDWR          0x03
#define O_CREAT         0x10
#define O_TRUNC         0x20
#define O_APPEND        0x40
#define O_EXCL          0x80
```
then these are non-UNIX values too. And finally: If there should be one day an 8-bit target actually supporting the mode it would be beneficial to only use 8-bit values allowing the 6502 open/creat code to ignore the high byte - just like with the flags above which are only 8-bit values although the data type is `int`.
So I would just go for
```
#define S_IREAD        0x01
#define S_WRITE        0x02
```